From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: [PATCH] add EXPORT_SYMBOL_GPL_FUTURE() to USB subsystem
Patch-mainline: 2.6.17

The USB core symbols will be converted to GPL-only in a few years.  Mark
this as such and update the documentation explaining why, and provide a
pointer for developers to receive help if they need it.

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
 Documentation/feature-removal-schedule.txt |   19 +++++++++++++++++++
 drivers/usb/core/driver.c                  |    6 +++---
 2 files changed, 22 insertions(+), 3 deletions(-)

--- linux-2.6.15.orig/drivers/usb/core/driver.c	2006-03-10 08:29:30.000000000 -0800
+++ linux-2.6.15/drivers/usb/core/driver.c	2006-03-10 08:35:46.000000000 -0800
@@ -378,7 +378,7 @@
 
 	return NULL;
 }
-EXPORT_SYMBOL(usb_match_id);
+EXPORT_SYMBOL_GPL_FUTURE(usb_match_id);
 
 int usb_device_match(struct device *dev, struct device_driver *drv)
 {
@@ -446,7 +446,7 @@
 
 	return retval;
 }
-EXPORT_SYMBOL(usb_register_driver);
+EXPORT_SYMBOL_GPL_FUTURE(usb_register_driver);
 
 /**
  * usb_deregister - unregister a USB driver
@@ -469,4 +469,4 @@
 
 	usbfs_update_special();
 }
-EXPORT_SYMBOL(usb_deregister);
+EXPORT_SYMBOL_GPL_FUTURE(usb_deregister);
--- linux-2.6.15.orig/Documentation/feature-removal-schedule.txt	2006-03-10 08:29:33.000000000 -0800
+++ linux-2.6.15/Documentation/feature-removal-schedule.txt	2006-03-10 08:36:03.000000000 -0800
@@ -189,3 +189,22 @@
 	users have complained indicating there is no more need for these
 	boards.  This should really be considered a last call.
 Who:	Ralf Baechle <ralf@linux-mips.org>
+
+---------------------------
+
+What:	USB driver API moves to EXPORT_SYMBOL_GPL
+When:	Febuary 2008
+Files:	include/linux/usb.h, drivers/usb/core/driver.c
+Why:	The USB subsystem has changed a lot over time, and it has been
+	possible to create userspace USB drivers using usbfs/libusb/gadgetfs
+	that operate as fast as the USB bus allows.  Because of this, the USB
+	subsystem will not be allowing closed source kernel drivers to
+	register with it, after this grace period is over.  If anyone needs
+	any help in converting their closed source drivers over to use the
+	userspace filesystems, please contact the
+	linux-usb-devel@lists.sourceforge.net mailing list, and the developers
+	there will be glad to help you out.
+Who:	Greg Kroah-Hartman <gregkh@suse.de>
+
+---------------------------
+
